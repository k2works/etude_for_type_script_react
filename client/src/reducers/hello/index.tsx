import { combineReducers } from 'redux';
import enthusiasm from './enthusiasm';

export default combineReducers({
    enthusiasm
});
