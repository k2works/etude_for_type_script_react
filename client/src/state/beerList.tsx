export interface IState {
    beers: string[];
    text: string;
}
