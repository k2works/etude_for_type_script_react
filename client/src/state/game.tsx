export interface IState {
    history:IHistoryData[];
    stepNumber:number;
    xIsNext:boolean;
}

export interface IHistoryData {
    squares: Array<('O' | 'X' | null)>;
}

