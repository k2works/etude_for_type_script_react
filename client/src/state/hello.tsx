export interface IState {
    enthusiasm: IHelloState;
}

export interface IHelloState {
    languageName: string;
    enthusiasmLevel: number;
}
