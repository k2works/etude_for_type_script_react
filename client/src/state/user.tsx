export interface IState {
    readonly users: IUser[];
}

export interface IUser {
    id: number;
    email: string;
    password: string;
}
