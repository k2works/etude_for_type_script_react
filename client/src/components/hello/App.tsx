import * as React from 'react'
import Hello from '../../containers/Hello'

const HelloApp: React.SFC = () => (
    <div>
        <Hello/>
    </div>
);
â€‹
export default HelloApp
