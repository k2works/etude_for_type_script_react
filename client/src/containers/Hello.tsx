import { connect, Dispatch } from "react-redux";
import { withRouter } from 'react-router-dom'
import * as actions from '../actions/';
import Hello from '../components/hello/Hello';
import  { IState } from "../state/hello";

export function mapStateToProps({ enthusiasm }: IState) {
    return {
        enthusiasm
    }
}

export function mapDispatchToProps(dispatch: Dispatch<actions.EnthusiasmAction>) {
    return {
        onIncrement: () => dispatch(actions.incrementEnthusiasm()),
        // tslint:disable-next-line:object-literal-sort-keys
        onDecrement: () => dispatch(actions.decrementEnthusiasm()),
    }
}

export default withRouter<any>(connect<any>(mapStateToProps,mapDispatchToProps)(Hello));
