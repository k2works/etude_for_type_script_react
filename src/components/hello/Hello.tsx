import * as React from 'react';
import {IHelloState} from "../../state/hello";
import './Hello.css';

interface IOwnProps {
    enthusiasm: IHelloState
    onIncrement?: () => void;
    onDecrement?: () => void;
}

function Hello( {enthusiasm, onIncrement, onDecrement }: IOwnProps ) {
    if (enthusiasm.enthusiasmLevel <= 0) {
        throw new Error('Your could be a little more enthusiastic. :D');
    }

    return (
        <div className="hello">
            <div className="greeting">
            Hello {enthusiasm.languageName + getExclamationMarks(enthusiasm.enthusiasmLevel)}
            </div>
            <div>
                <button onClick={onDecrement}>-</button>
                <button onClick={onIncrement}>+</button>
            </div>
        </div>
    );
}

export default Hello;

// helpers

function getExclamationMarks(numChars: number) {
    return Array(numChars + 1).join('!');
}
