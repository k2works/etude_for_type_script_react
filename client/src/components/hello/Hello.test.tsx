import * as enzyme from 'enzyme';
import * as React from 'react';
import * as renderer from "react-test-renderer";
import {IHelloState} from "../../state/hello";
import Hello from './Hello';

it('renders the correct text with an explicit enthusiasm of 1', () => {
    const initialState: IHelloState = {languageName:'Daniel',enthusiasmLevel:1};
    const hello = enzyme.shallow(<Hello enthusiasm={initialState} /> );
    expect(hello.find(".greeting").text()).toEqual('Hello Daniel!');
});

it('renders the correct text with an explicit enthusiasm level of 5', () => {
    const initialState: IHelloState = {languageName:'Daniel',enthusiasmLevel:5};
    const hello = enzyme.shallow(<Hello enthusiasm={initialState} /> );
    expect(hello.find(".greeting").text()).toEqual('Hello Daniel!!!!!');
});

it('throw when then enthusiasm level is 0', () => {
    const initialState: IHelloState = {languageName:'Daniel',enthusiasmLevel:0};
    expect(() => {
        enzyme.shallow(<Hello enthusiasm={initialState} /> );
    }).toThrow();
});

it('throws when the enthusiasm level is negative,', () => {
    const initialState: IHelloState = {languageName:'Daniel',enthusiasmLevel:-1};
    expect(() => {
        enzyme.shallow(<Hello enthusiasm={initialState} />);
    }).toThrow();
});

it('renders correctly', () => {
    const initialState: IHelloState = {languageName:'Daniel',enthusiasmLevel:10};
    const tree = renderer
        .create(<Hello enthusiasm={initialState} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
