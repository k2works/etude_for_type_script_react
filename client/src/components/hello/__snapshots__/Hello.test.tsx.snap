// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="hello"
>
  <div
    className="greeting"
  >
    Hello 
    Daniel!!!!!!!!!!
  </div>
  <div>
    <button
      onClick={undefined}
    >
      -
    </button>
    <button
      onClick={undefined}
    >
      +
    </button>
  </div>
</div>
`;
